whitepapers/0013_ccq.md

```diff
@@ -0,0 +1,350 @@
+
+### Request Validation
+
+The query module will perform request validation consisting of the following:
+
+- Verify the signature. This includes verifying the request was signed with the right prefix, so that requests from other environments can't be replayed
+  (for example, a testnet message cannot be played in mainnet).
+- Verify that the signer is allowed to execute CCQ requests (is allow listed).
+- Verify that all of the per-chain requests contained in the query are valid.
+
+#### Block ID in eth_call Queries
+
+Note that for `eth_call` queries (the only type currently supported), the `block_id` must be either a block number or block hash. Tags like `latest` or `finalized` are not supported. This is because different guardians may well have a different value for either `latest` or `finalized`, depending on the
+state of their nodes.
+
+Note that there may be a need to support the use of tags like `latest` and `finalized`, which may require gossiping block numbers or having the query server read the data. This will be handled as a follow on feature.
+
+#### Signature Verification
+
+Requests messages MUST include a signature in the payload in order to distinguish between a requestor and (potentially, third-party) p2p relayer.
+That signature SHOULD be created using a separate key.
+
+In order to differentiate signatures and prevent replay attacks of requests intended for devnet/testnet from mainnet ones, the following prefixes MUST be used when signing. These prefix strings are padded to 35-bytes to align with [Guardian Key Usage](https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0009_guardian_key.md) and existing code.
+
+```
+mainnet_query_request_000000000000|
+testnet_query_request_000000000000|
+devnet_query_request_0000000000000|
+
+```
+
+### Request Execution
+
+Once the request has been validated, the query module will submit the individual per-chain query requests to the appropriate watchers for execution. The watchers will submit the RPC calls
+to the node and return the results. Communication between the query module and the watchers is via a pair of golang channels per watcher, one inbound and one outbound. The watchers will
+use batch requests to minimize RPC overhead. For this to work effectively, the integrator should properly group RPC calls into the minimal set of per-chain queries.
+
+The query module will listen for responses for all of the per-chain queries. When all per-chain responses are received, the module will post the result to be published on the gossip network.
+If any of the responses fails or times out, the query module will retry periodically for up to one minute. If after a minute some of the per-chain queries were not successful, the query
+module will drop the request.
```

Comment:
We chose not to respond with an error because that opens up an DoS vector. A single bad request could generate 19 error responses, allowing a malicious actor to flood the gossip network. The CCQ query server does do validation and return an error.

I updated the text to explain this.